client/src/App.js

```diff
@@ -0,0 +1,58 @@
+import React, { Component } from 'react';
+import Dashboard from './components/Dashboard';
+import Display from './components/Display';
+import './App.css';
+
+class App extends Component {
+  state = {
+    balls: 0,
+    strikes: 0
+  }
+
+  ballCount = e => {
+    let count = this.state.balls + 1;
+    if(count === 4){
+      this.setState({balls: 0});
+    } else {
+      this.setState({balls: count})
+    }
+  };
+
+  strikeCount = e => {
+    let count = this.state.strikes + 1;
+    if(count === 3){
+      this.setState({strikes: 0});
+    } else {
+      this.setState({strikes: count});
+    }
+  };
+
+  hitCount = e => {
+    this.setState({
+      balls: 0,
+      strikes: 0
+    });
+  };
+
+  foulCount = e => {
+    if(this.state.strikes === 2){
+      return;
+    }
+    let count = this.state.strikes + 1;
+    this.setState({strikes: count});
+  };
+
+  render() {
+    return (
+      <div className="App">
+        <Display balls={this.state.balls} strikes={this.state.strikes}/>
+        <Dashboard ballCount={this.ballCount}
+                   strikeCount={this.strikeCount}
+                   hitCount={this.hitCount}
+                   foulCount={this.foulCount}/>
```

Comment:
I image you could iterate through these counts in order to pass them; i.e., at the top of your render:
```javascript
const countMethods = [ 'ball', 'strike', 'hit', 'foul' ]
  .reduce((counts, data) => {
    const countMethod = `${data}Count`;
    return { ...counts, [countMethod]: this[countMethod] };
  }, {});

// ... later, in your return, just under Display:
<Dashboard { ...countMethods } />
```
This is largely an exercise for passing similar props like this on an even larger scale, but hopefully it's demonstrative regardless!